Update on-call engineer handbook entries (#38641)
fleetdm/fleet: handbook/engineering/README.md

After-hours incident on-call engineers are available outside the business hours of 9am - 6pm Pacific. The [on-call support SLA](https://fleetdm.com/handbook/company/product-groups#business-hours-on-call-responsibilities) requires a 1-hour response time outside business hours to any `@incident-oncall` mention.

#### Assume after hours on-call engineer alias

After-hours incident on-call engineer rotation, alias assignment, and incident notification are managed through incident.io and reported in the #help-incidents channel. 

The on-call engineer is responsible for: 

- Knowing [the after-hours incidenton-call rotation](https://fleetdm.com/handbook/company/product-groups#after-hours-incident-on-call).
- Completing the [incident.io on-call engineer onboarding steps](https://help.incident.io/articles/3472064049-get-started-as-an-on-call-responder) sent via email when invited to incident.io.

> - Confirming the app is configured to push notifications through silent mode.

Reviewer: ```suggestion
- Confirming the app is configured to push notifications through Do Not Disturb.
```
